paimon-common/src/main/java/org/apache/paimon/sst/SstFileReader.java

```diff
@@ -50,6 +54,7 @@ public class SstFileReader implements Closeable {
     private final BlockCache blockCache;
     private final BlockIterator indexBlockIterator;
```

Comment:
Can you introduce a `BlockMeta` for this? Just store data, recordCount, comparator.